Source/WebCore/animation/TimelineRange.cpp

```diff
@@ -43,11 +48,120 @@ bool SingleTimelineRange::isDefault(const CSSPrimitiveValue& value, Type type)
+{
+    if (!value || value->isCalculated() || !element)
+        return { };
+    if (value->valueID() == CSSValueAuto)
+        return { };
+
+    auto conversionData = cssToLengthConversionData(element);
+    if (!conversionData) {
+        if (value->isPercentage())
+            return Length(value->resolveAsPercentageNoConversionDataRequired(), LengthType::Percent);
+        if (value->isPx())
+            return Length(value->resolveAsLengthNoConversionDataRequired(), LengthType::Fixed);
+        return { };
+    }
+
+    if (value->isLength())
+        return value->resolveAsLength<WebCore::Length>(*conversionData);
+
+    if (value->isPercentage())
+        return WebCore::Length(value->resolveAsPercentage(*conversionData), LengthType::Percent);
+
+    if (value->isCalculatedPercentageWithLength())
+        return WebCore::Length(value->cssCalcValue()->createCalculationValue(*conversionData, CSSCalcSymbolTable { }));
+
+    ASSERT_NOT_REACHED();
+    return { };
+}
+
+SingleTimelineRange SingleTimelineRange::range(const CSSValue& value, Type type, const Style::BuilderState* state, RefPtr<Element> element)
+{
+    if (RefPtr primitiveValue = dynamicDowncast<CSSPrimitiveValue>(value)) {
+        if (SingleTimelineRange::isOffsetValue(*primitiveValue)) {
+            // <length-percentage>
+            return { SingleTimelineRange::Name::Omitted, state ? Style::BuilderConverter::convertLength(*state, *primitiveValue) : lengthFoCSSValue(primitiveValue, element) };
+        }
+        // <timeline-range-name> or Normal
+        return { SingleTimelineRange::timelineName(primitiveValue->valueID()), defaultValue(type) };
+    }
+    RefPtr pair = dynamicDowncast<CSSValuePair>(value);
+    if (!pair)
+        return { };
+
+    // <timeline-range-name> <length-percentage>
+    Ref primitiveValue = downcast<CSSPrimitiveValue>(pair->second());
+    ASSERT(SingleTimelineRange::isOffsetValue(primitiveValue));
+
+    return { SingleTimelineRange::timelineName(pair->first().valueID()), state ? Style::BuilderConverter::convertLength(*state, primitiveValue) : lengthFoCSSValue(RefPtr { primitiveValue.ptr() }, element) };
+}
+
+SingleTimelineRange SingleTimelineRange::parse(TimelineRangeValue&& value, RefPtr<Element> element, Type type)
+{
+    if (!element)
+        return { };
+    const auto& parserContext = element->document().cssParserContext();
+    return WTF::switchOn(value,
+    [&](String& rangeString) {
+        CSSTokenizer tokenizer(rangeString);
+        auto tokenRange = tokenizer.tokenRange();
+        tokenRange.consumeWhitespace();
+        if (auto consumedRange = type == SingleTimelineRange::Type::Start ? CSSPropertyParserHelpers::consumeAnimationRangeStart(tokenRange, parserContext) : CSSPropertyParserHelpers::consumeAnimationRangeEnd(tokenRange, parserContext))
+            return range(*consumedRange, type, nullptr, element);
+        return SingleTimelineRange { };
```

Comment:
Might be wrong but you possibly don't need `SingleTimelineRange` here?

Reply:
This is necessary to fix build issues.